torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,390 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Cloud Cover Detection Challenge dataset."""
+
+import json
+import os
+from typing import Any, Callable, Dict, List, Optional, Sequence
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from numpy.typing import NDArray
```

Comment:
This module isn't available in older versions of numpy, which is why the tests are failing. If you instead remove this import and use `"np.typing.NDArray[...]"` (note the quotes) then the import is only evaluated when you run `mypy`, which we only run with the latest version of numpy.